lib/apis/v3/felixconfig.go

```diff
@@ -265,6 +265,35 @@ type FelixConfigurationSpec struct {
 	// modes can use XDP. This is not recommended since it doesn't provide better performance than
 	// iptables. [Default: false]
 	GenericXDPEnabled *bool `json:"genericXDPEnabled,omitempty" confignamev1:"GenericXDPEnabled"`
+
+	// BPFEnabled, if enabled Felix will use the BPF dataplane. [Default: false]
+	BPFEnabled *bool `json:"bpfEnabled,omitempty" validate:"omitempty,"`
+	// BPFLogLevel controls the log level of the BPF programs when in BPF dataplane mode.  One of "Off", "Info", or
+	// "Debug". [Default: Off].
+	BPFLogLevel string `json:"bpfLogLevel" validate:"omitempty,bpfLogLevel"`
+	// BPFDataIfacePattern is a regular expression that controls which interfaces Felix should attach BPF programs to
+	// in order to catch traffic to/from the network.  This needs to match the interfaces that calico workload traffic
+	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the
+	// cluster.  It should not match the workload interfaces (usually named cali...).  [Default: ^(en.*|eth.*|tunl0$)]
+	BPFDataIfacePattern string `json:"bpfDataIfacePattern,omitempty" validate:"omitempty,regexp"`
+	// BPFConnectTimeLoadBalancingEnabled when in BPF mode, controls whether Felix installs the connection-time load
+	// balancer.  The connect-time load balancer is required for the host to be able to reach Kubernetes services
+	// and it improves the performance of pod-to-service connections.  The only reason to disable it is for debugging
+	// purposes.  [Default: true]
```

Comment:
If debugging-only, do we really need it in the data model here?

Reply:
Think we may support running without this enabled more thoroughly in future so I put it just on this side of the debug-only line.